### DISCUSSION TITLE
Loss function and masking

### DISCUSSION THREAD

**Discussion by RostislavStoyanov** [amazon-science/chronos-forecasting#482] [Q&A]
Hi,

I have a question regarding the loss computation for Chronos2,when training with covariates. From my understanding,  covariates are multiplied with a mask to stop them from impacting loss calculation. However, if we look at the final loss computation:
https://github.com/amazon-science/chronos-forecasting/blob/6d68ed7c4ed2805d122d77b4660765b4089de5ca/src/chronos/chronos2/model.py#L546
The last mean is taken over the whole batch. I am wondering if this is correct as I think it can lead to artificial loss decrease in cases where we introduce new covariates.
For example, we finetune a model with 5 covariates and we get losses in certain range. We then introduce 5 more covariates which are not very informative, so the model learns to ignore them and we expect the loss to stay roughlt in the same range. The number of targets in a batch of the same size would decrease, which would lead to the numerator being smaller while the denominator stays the same, so the loss naturally gets smaller. This would make loss values comparison between the two runs invalid. Wouldn't a more correct approach be to calculate the mean over the number of targets that are actually included in the loss computation or is there a reason for the current approach?
